docs/general/builtins/table-functions.rst

```diff
@@ -198,6 +198,152 @@ arrays within the same level.
+are given as a string containing any of the characters listed below. Order does
+not matter.
+
++-------+---------------------------------------------------------------------+
+| Flag  | Description                                                         |
++=======+=====================================================================+
+| ``i`` | enable case insensitive matching                                    |
++-------+---------------------------------------------------------------------+
+| ``u`` | enable unicode case folding when used together with ``i``           |
++-------+---------------------------------------------------------------------+
+| ``U`` | enable unicode support for character classes like ``\W``            |
++-------+---------------------------------------------------------------------+
+| ``s`` | make ``.`` match line terminators, too                              |
++-------+---------------------------------------------------------------------+
+| ``m`` | make ``^`` and ``$`` match on the beginning or end of a line        |
+|       | too.                                                                |
++-------+---------------------------------------------------------------------+
+| ``x`` | permit whitespace and line comments starting with ``#``             |
++-------+---------------------------------------------------------------------+
+| ``d`` | only ``\n`` is considered a line-terminator when using ``^``, ``$`` |
+|       | and ``.``                                                           |
++-------+---------------------------------------------------------------------+
+| ``g`` | keep matching until the end of ``source``, instead of stopping at   |
+|       | the first match.                                                    |
++-------+---------------------------------------------------------------------+
+
+
+Examples
+........
+
+::
+
+    cr> select regexp_matches('foobar', '^(a(.+)z)$') as matched;
+    +---------+
+    | matched |
+    +---------+
+    +---------+
+    SELECT 0 rows in set (... sec)
+
+::
+
+    cr> select regexp_matches('99 bottles of beer on the wall', '\d{2}\s(\w+).*', 'ixU')
+    ... as matched;
+    +-------------+
+    | matched     |
+    +-------------+
+    | ["bottles"] |
+    +-------------+
+    SELECT 1 row in set (... sec)
+
+::
+
+    cr>  select regexp_matches('#abc #def #ghi #jkl', '(#[^\s]*) (#[^\s]*)', 'g') as matched;
+    +------------------+
+    | matched          |
+    +------------------+
+    | ["#abc", "#def"] |
+    | ["#ghi", "#jkl"] |
+    +------------------+
+    SELECT 2 rows in set (... sec)
+
+::
+
+    cr> select regexp_matches('foobarbequebazilbarfbonk', '(b[^b]+)(b[^b]+)', 'g') as matched;
+    +-------------------+
+    | matched           |
+    +-------------------+
+    | ["bar", "beque"]  |
+    | ["bazil", "barf"] |
+    +-------------------+
+    SELECT 2 rows in set (... sec)
```

Comment:
Same here as in the generate_subscripts PR. I think for users it would be helpful to have a sentence or two for each example, explaining what is shown.

I'm also not sure what the difference between the last two examples is? Feels a bit like proofing that it works instead of illustrating the user how it can be used.